src/pages/index.js

```diff
@@ -20,100 +24,212 @@ import {
+/*-- Trash Icon Selectors --*/
+const trashModal = document.querySelector("#trash-modal");
+const trashModalSubmitButton = trashModal.querySelector(".modal__button");
 
 /*---------------------------------------------------*/
-/*                     Functions                     */
+/*                 Event Handlers                    */
 /*---------------------------------------------------*/
 
-function createCard(cardData) {
-  const name = cardData.name;
-  const altName = cardData.name;
-  const link = cardData.link;
-  const card = new Card(
-    { name, altName, link },
-    "#card-template",
-    handleImageClick
-  );
-  return card.getView();
+function handleAvatarFormSubmit(formValues) {
+  api
+    .patchProfileAvatar(formValues.url)
+    .then((object) => {
+      changeSubmitButton(avatarModalSubmitButton);
+      avatarPic.src = object.avatar;
+      avatarPopup.close();
+      profileAvatarForm.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileAvatarFormValidator.disableButton();
+      // avatarModalSubmitButton.textContent = "Save";
+    });
 }
 
-/*---------------------------------------------------*/
-/*                 Event Handlers                    */
-/*---------------------------------------------------*/
-
 function handleProfileFormSubmit(formValues) {
-  userInfo.setUserInfo(formValues.name, formValues.bio);
-  profilePopup.close();
-  editProfileFormValidator.disableButton();
+  api
+    .patchProfile(formValues.name, formValues.bio)
+    .then(() => {
+      changeSubmitButton(editModalSubmitButton);
+      userInfo.setUserInfo(formValues.name, formValues.bio);
+      // profilePopup.renderLoading(true);
+      profilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      // profilePopup.renderLoading(false);
+      editProfileFormValidator.disableButton();
+      // editModalSubmitButton.textContent = "Save";
+    });
 }
 
 function handleAddCardFormSubmit(formValues) {
   const name = formValues.title;
-  const altName = formValues.title;
   const link = formValues.url;
-  const cardElement = createCard({ name, altName, link });
-  section.addItem(cardElement);
-  cardPopup.close();
+  api
+    .postCards({ name, link })
+    .then((object) => {
+      changeSubmitButton(addModalSubmitButton);
+      const element = createCard(object);
+      section.addItem(element);
+      cardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      // addModalSubmitButton.textContent = "Create";
+    });
 }
 
 function handleImageClick(data) {
   popupImg.open(data);
 }
 
+function handleConfirmModal(data) {
+  trashConfirmPopup.open();
+  trashConfirmPopup.setSubmitFunction(() => {
+    handleDeleteConfirmModal(data.apiData._id);
+    data.removeCard(data.cardElement);
+  });
+}
+
+function handleDeleteConfirmModal(ID) {
+  cards.then((cards) => {
```

Comment:
When you delete a cards, you don't need to `loop through` the cards. You set the click handler inside `Card`, so you have a reference for the card element